src/main/java/de/zalando/aruha/nakadi/service/EventPublisher.java

```diff
@@ -50,38 +50,37 @@ public EventPublisher(final TopicRepository topicRepository,
     public EventPublishResult publish(final JSONArray events, final String eventTypeName) throws NoSuchEventTypeException,
             InternalNakadiException {
         final EventType eventType = eventTypeCache.getEventType(eventTypeName);
-        final List<BatchItem> batch = initBatch(events);
+        final List<BatchItem> batch = BatchFactory.from(events);
 
         try {
             validate(batch, eventType);
-            enrich(batch, eventType);
             partition(batch, eventType);
+            enrich(batch, eventType);
```

Comment:
@v-stepanov well observed. I think we should change the API as well https://github.com/zalando/nakadi/blob/nakadi-jvm/api/nakadi-event-bus-api.yaml#L1115 since some users might expect this order to be well defined. There used to be no dependency between these two steps but now it does.